cmd/commands/cmd_import_mission_control.go

```diff
@@ -98,3 +101,161 @@ func importMissionControl(ctx *cli.Context) error {
+func loadMissionControl(ctx *cli.Context) error {
+	rpcCtx := context.Background()
+
+	mcDataPath := ctx.String("mcdatapath")
+	if mcDataPath == "" {
+		return fmt.Errorf("mcdatapath must be set")
+	}
+
+	if _, err := os.Stat(mcDataPath); os.IsNotExist(err) {
+		return fmt.Errorf("%v does not exist", mcDataPath)
+	}
+
+	conn := getClientConn(ctx, false)
+	defer conn.Close()
+
+	client := routerrpc.NewRouterClient(conn)
+
+	// Load and unmarshal the querymc output file.
+	mcRaw, err := os.ReadFile(mcDataPath)
+	if err != nil {
+		return fmt.Errorf("could not read querymc output file: %w", err)
+	}
+
+	mc := &routerrpc.QueryMissionControlResponse{}
+	err = lnrpc.ProtoJSONUnmarshalOpts.Unmarshal(mcRaw, mc)
+	if err != nil {
+		return fmt.Errorf("could not unmarshal querymc output file: %w",
+			err)
+	}
+
+	// We discard mission control data if requested.
+	if ctx.Bool("discard") {
+		if !promptForConfirmation("This will discard mission control " +
+			"data: (yes/no)") {
+
+			return nil
+		}
+
+		_, err = client.ResetMissionControl(
+			rpcCtx, &routerrpc.ResetMissionControlRequest{},
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	// Add a time offset to all timestamps if requested.
+	timeOffset := ctx.String("timeoffset")
+	if timeOffset != "" {
+		offset, err := time.ParseDuration(timeOffset)
+		if err != nil {
+			return fmt.Errorf("could not parse time offset: %w",
+				err)
+		}
+
+		var maxTimestamp time.Time
+
+		for _, pair := range mc.Pairs {
+			if pair.History.SuccessTime != 0 {
+				unix := time.Unix(pair.History.SuccessTime, 0)
+				unix = unix.Add(offset)
+
+				if unix.After(maxTimestamp) {
+					maxTimestamp = unix
+				}
+
+				pair.History.SuccessTime = unix.Unix()
+			}
+
+			if pair.History.FailTime != 0 {
+				unix := time.Unix(pair.History.FailTime, 0)
+				unix = unix.Add(offset)
+
+				if unix.After(maxTimestamp) {
+					maxTimestamp = unix
+				}
+
+				pair.History.FailTime = unix.Unix()
+			}
+		}
+
+		fmt.Printf("Adding time offset %v to all timestamps. "+
+			"New max timestamp: %v\n", offset, maxTimestamp)
+	}
+
+	sanitizeMCData(mc.Pairs)
+
+	fmt.Printf("Mission control file contains %v pairs.\n", len(mc.Pairs))
+	if !promptForConfirmation("Import mission control data (yes/no):") {
+		return nil
+	}
+
+	_, err = client.XImportMissionControl(
+		rpcCtx,
+		&routerrpc.XImportMissionControlRequest{
+			Pairs: mc.Pairs,
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("could not import mission control data: %w",
+			err)
+	}
+
+	return nil
+}
+
+// sanitizeMCData removes invalid data from the exported mission control data.
+func sanitizeMCData(mc []*routerrpc.PairHistory) {
+	for _, pair := range mc {
+		// It is not allowed to import a zero-amount success to mission
+		// control if a timestamp is set. We unset it in this case.
+		if pair.History.SuccessTime != 0 &&
+			pair.History.SuccessAmtMsat == 0 &&
+			pair.History.SuccessAmtSat == 0 {
+
+			pair.History.SuccessTime = 0
+		}
```

Comment:
When testing I've seen entries like that
```
        {
            "node_from":  "X",
            "node_to":  "Y",
            "history":  {
                "fail_time":  "1746388491",
                "fail_amt_sat":  "0",
                "fail_amt_msat":  "0",
                "success_time":  "1746386085",
                "success_amt_sat":  "0",
                "success_amt_msat":  "0"
            }
        },
```
which is inconsistent with what we allow to import, which is why I've added those clean-ups. See here:

https://github.com/lightningnetwork/lnd/blob/334a7d112378276229b5e96b8642e16204278f1e/lnrpc/routerrpc/router_server.go#L1282-L1286